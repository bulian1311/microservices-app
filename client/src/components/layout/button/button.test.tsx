import * as React from "react";
import { render, RenderResult, fireEvent } from "@testing-library/react";
import Button from "./button.component";

let documentBody: RenderResult;

beforeEach(() => {
  documentBody = render(<Button>Test button</Button>);
});

test("Button matces snapshot.", () => {
  const { baseElement } = documentBody;
  expect(baseElement).toMatchSnapshot();
});

test("Button inner text.", () => {
  expect(documentBody.getByText("Test button")).toBeInTheDocument();
});

test("Button on click event.", () => {
  let str: string = "";

  const { getByTestId } = render(
    <Button testId="btn" onClick={() => (str = "btn")}>
      Test button
    </Button>
  );

  const btn = getByTestId("btn");

  fireEvent.click(btn);

  expect(str).toBe("btn");
});
