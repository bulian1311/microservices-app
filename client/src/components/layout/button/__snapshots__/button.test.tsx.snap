// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matces snapshot. 1`] = `
<body>
  <div>
    <a
      class="sc-bdfBwQ kPJUTy"
    >
      Test button
    </a>
  </div>
</body>
`;
